src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2271,12 +2287,12 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
         {
             UnaryOperatorKind kind = SyntaxKindToUnaryOperatorKind(node.Kind());
 
-            bool isOperandTypeNull = operand.IsLiteralNull() || operand.IsLiteralDefault();
+            bool isOperandTypeNull = operand.IsLiteralNull() || operand.IsLiteralDefault() || operand.IsTypelessNew();
             if (isOperandTypeNull)
```

Comment:
Please add a PROTOTYPE comment that this should be rewritten in terms of a test of the type against `null`.